torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -730,6 +730,7 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       at::cuda::getNumGPUs() != 0,
       "ProcessGroupNCCL is only supported with GPUs, no GPUs found!");
   this->setGroupName(options_->group_name);
+  this->setGroupDesc(options_->group_desc);
```

Comment:
Hmm, why are we setting the desc so low in the stack? It would seem only functional to NCCL.
Is it possible to set the desc at the python layer of `ProcessGroup`? (not `ProcessGroupNCCL`)

Reply:
Here we want to know the group desc as well when creating ProcessGroupNCCL, for example, print set up, create log prefix, profiler, etc thus in processgroupNCCL we have this set logic. However, we do set the desc at python layer with pg. _set_group_desc after the pg creation so all backends will have the desc as well.

Reply:
> The Backend class will know it bc ProcessGroup set its desc to each backend

But ProcessGroup set name/desc to each backend class **after** the initialization of a backend. In order to know the group_desc during the initialization of backend(ProcessGroupNCCL), we have to change ProcessGroupNCCL constructor. I think this will need more changes, especially some user code that calling `dist.ProcessGroupNCCL` not through c10d `new_group`.

Reply:
See my other comment:
> why must we access this information in ProcessGroupNCCL's init function? For printing a log there? Not critical to me.

Reply:
> Not only printing a log, but also pass to profiler, and flight recorder to know pg set up.